crypto/rand/drbg_lib.c

```diff
@@ -359,9 +352,15 @@ int RAND_DRBG_instantiate(RAND_DRBG *drbg,
     }
 
     drbg->state = DRBG_READY;
-    drbg->reseed_gen_counter = 1;
+    drbg->generate_counter = 1;
     drbg->reseed_time = time(NULL);
-    tsan_store(&drbg->reseed_prop_counter, drbg->reseed_next_counter);
+    if (drbg->enable_reseed_propagation) {
+        if (drbg->parent == NULL)
+            tsan_counter(&drbg->reseed_counter);
+        else
+            tsan_store(&drbg->reseed_counter,
```

Comment:
I don't understand why `tsan_store` is required since only this thread accesses this counter.  Still, it's harmless to have it here.

Reply:
I also pondered whether it is necessary, but then I left it because I had the case in mind where the child DRBG itself is accessed concurrently by "grandchildren" DRBGs.

This does not occur in our standard DRBG triple setup, but might be a setup in some application. (The new EVP_RAND does support chaining by the application, doesn't it?) Of course the application needs to guard concurrent access to the child DRBG (in the middle of the chain) by a lock, but that does not guarantee an atomic write operation of the reseed_counter, if the grandchildren use only atomics with relaxed semantics and not a read lock (of an rwlock). At least that's how I understood @bernd-edlinger's explanations.

Reply:
Good point.